### ISSUE TITLE
tkinter widgets should allow a 'name' parameter.

### ISSUE THREAD

**Issue by Athanasius** [python/typeshed#4658]
I'm working with some legacy code that includes lines like:

```py
self.cmdr = tk.Label(frame, compound=tk.RIGHT, anchor=tk.W, name='cmdr')
```
but mypy (0.790) complains:
```
EDMarketConnector.py:144: error: Unexpected keyword argument "name" for "Label"
```
This appears to be due to the lack of a 'name' parameter in `mypy\typeshed\stdlib\3\tkinter\__init__.py` for Label objects.  Similarly for at least Frame and Menu as well.

Setting a name should be allowed on any widget.  Tk will set one itself if none is specified (based on the widget type), but you can change this.

Using Python 3.8.6 and tracing things through, such a name parameter is made use of in `C:\Program Files (x86)\Python38-32\Lib\tkinter\__init__.py`:

```py
class BaseWidget(Misc):
    """Internal class."""

    def _setup(self, master, cnf):
        """Internal function. Sets up information about children."""
        if _support_default_root:
            global _default_root
            if not master:
                if not _default_root:
                    _default_root = Tk()
                master = _default_root
        self.master = master
        self.tk = master.tk
        name = None
        if 'name' in cnf:
            name = cnf['name']
            del cnf['name']
...
```

**Comment by srittau** [comment]
I have no experience with tkinter, but maybe @Akuli has an opinion? That said, we gladly accept PRs that improve typeshed.

**Comment by Akuli** [comment]
Can you create a PR that fixes this? It should be added to all widgets in `__init__.py` and `ttk.py`.

**Comment by melassa** [comment]
I do not see `BaseWidget.name` in Python 3.8.6, i see a `_name` field.

**Comment by srittau** [comment]
Closed in #4658.

**Comment by Akuli** [comment]
#4867  is the PR that fixed this
